crates/shared/src/price_estimation/competition.rs

```diff
@@ -260,47 +281,42 @@ impl PriceEstimating for CompetitionEstimator<Arc<dyn PriceEstimating>> {
     }
 }
 
-fn is_second_quote_result_preferred(
+fn compare_quote_result(
     query: &Query,
     a: &PriceEstimateResult,
     b: &PriceEstimateResult,
     context: &RankingContext,
-) -> bool {
+) -> Ordering {
     match (a, b) {
-        (Ok(a), Ok(b)) => is_second_estimate_preferred(query, a, b, context),
-        (Ok(_), Err(_)) => false,
-        (Err(_), Ok(_)) => true,
-        (Err(a), Err(b)) => is_second_error_preferred(a, b),
+        (Ok(a), Ok(b)) => compare_quote(query, a, b, context),
+        (Ok(_), Err(_)) => Ordering::Greater,
+        (Err(_), Ok(_)) => Ordering::Less,
+        (Err(a), Err(b)) => compare_error(a, b),
     }
 }
 
-fn is_second_native_result_preferred(
+fn compare_native_result(
     a: &Result<f64, PriceEstimationError>,
     b: &Result<f64, PriceEstimationError>,
-) -> bool {
+) -> Ordering {
     match (a, b) {
-        (Ok(a), Ok(b)) => b >= a,
-        (Ok(_), Err(_)) => false,
-        (Err(_), Ok(_)) => true,
-        (Err(a), Err(b)) => is_second_error_preferred(a, b),
+        (Ok(a), Ok(b)) => a.total_cmp(b).reverse(),
```

Comment:
```suggestion
        (Ok(a), Ok(b)) => b.total_cmp(a),
```

Reply:
While your version is more concise I think having the explicit `.reverse()` makes the difference more obvious and looks more intentional.

Reply:
Is this correct though? Before we would prefer the second result, if b >= a. This should result in `Ordering:Less` in the current implementation (second being better in terms of `maxBy`).

However `a.total_comp(b).reverse()` returns `OrderingLess` iff `a > b` or am I wrong?

Could we add a test case for this?